### DISCUSSION TITLE
Clarifications on motion detection - http, webhooks and polling

### DISCUSSION THREAD

**Discussion by thanpolas** [fwestenberg/reolink_dev#620] [Q&A]
Hello there, 

thank you for the amazing work of integrating reolink to HA!

I am considering my options on how to integrate my 6 RLC-820A cameras and have been reading on the documentation, issues and posts on the discussions. Here is my understanding so far:

* For motion detection a webhook is created by the integration
* The webhook by reolink only supports http but not https
* For the Person / Vehicle (AI) motion detection can only be polled.

My goal is to integrate person motion detection to HA, so my questions are aimed towards that goal:

1. Is it still the case that AI motion detection needs to be polled?
2. If so, how is this poll implemented and what is polled? (what's the polling interval, and which API is hit?)
3. Could we possibly create a work-around for the http restriction? Like setup a local forwarding service or something to similar effect?
4. I searched for reolink's API documentation and stumbled on a "PDF documentation" ... is this for real? Do they really not have a website with the API documentation? 🤯 
5. If my goal is not possible right now (no solution for webhook AI motion detection and http workaround) --> I want to fall back to controlling push notifications on my cameras. I noticed however, that HA is not able to control push notifications by calling the "Camera: Disable motion detection" on the reolink device I added... is this integration implemented? (note: my push notifications are configured to only trigger on person motion detection)

**Comment by cpainchaud** [comment]
Hi,

 AI detection will be triggered webhooks BUT it many cases I have seen the webhooks fail for so many reasons (i am openly blaming reolink for poor software quality) that I use polling every 3 seconds with great success and 0 miss.

This is how you disable webhooks and then lower the polliong internal delay:

![image](https://user-images.githubusercontent.com/6696638/215421598-1264f103-780e-4936-98e3-ffcd40c1be0a.png)


 To answer your questions on API docs : yes it's as ridiculous as their software quality.
